docs/hooks.md

```diff
@@ -0,0 +1,127 @@
+# Hooks
+
+UIx wraps existing React hooks to cut some corners and provide a more idiomatic interafce for Clojure. `uix.core` exposes only default React hooks, named against their JS counterparts but in kebab-case.
+
+There are multiple differences from pure React though.
+
+## Dependency array
+
+Some hooks accept an array of dependencies as the second argument. While in pure React it has to be an array literal `#js []`, in UIx as Clojure library the deps are a vector literal `[]`.
+
+```clojure
+(uix/use-effect
+  (fn [] (prn x))
+  [x])
+```
+
+## How depedencies are compared?
+
+When the same depency has a different value between component updates, a hook will re-run.
+
+```clojure
+;; 1st update
+(uix/use-effect
+  (fn [] (prn x)) ;; prn 1
+  [x]) ;; x = 1
+
+;; 2nd update
+(uix/use-effect
+  (fn [] (prn x))
+  [x]) ;; x = 1, didn't change, do nothing
+
+;; 3rd update
+(uix/use-effect
+  (fn [] (prn x)) ;; prn 2
+  [x]) ;; x = 2, `x` has changed, re-run the hook
+```
+
+That works as expected for primitive values like numbers and strings, but what about Clojure's maps and vectors that can be compared by value `(= {:x 1} {:x 1}) ;; true`?
+
+Two maps equal by value but with different identity are considered to be not equal when React compares dependencies. This is happening because React is JS library that is not aware of Clojure's equality algorithm, instead React is comparing values by identity `===` or `Object.is`. Thus it's important to think about what type of values you are passing as dependencies of a hook. It applies to JS as well, objects and arrays are still compared by reference.
```

Comment:
```suggestion
Two maps equal by value but with different identity are considered to be not equal when React compares dependencies. This is happening because React is a JS library that is not aware of Clojure's equality algorithm, instead React is comparing values by identity `===` or `Object.is`. Thus it's important to think about what type of values you are passing as dependencies of a hook. It applies to JS as well, objects and arrays are still compared by reference.
```